app/components/youtube/model.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const _get = require('lodash/get'),
+  { getVideoDetails } = require('../../services/universal/youtube'),
+  defaultPlayerBorderTopCTA = 'Watch';
+
+/**
+ * Override various settings by type of video
+ * @param {Object} data
+ */
+function updateSettingsByType(data) {
+  switch (data.videoType) {
+    case 'related':
+      // By default, display borders and CTA when `related` type is first selected, afterwards accept user's selection
+      data.playerBorderTopCTA = !data.previousTypeRelated && !data.playerBorderTopCTA ? defaultPlayerBorderTopCTA : data.playerBorderTopCTA;
+      data.playerBorderTop = !data.previousTypeRelated ? true : data.playerBorderTop;
+      data.playerBorderBottom = !data.previousTypeRelated ? true : data.playerBorderBottom;
+      data.previousTypeRelated = true;
+      break;
+    case 'sponsored':
+      data.autoPlay = false;
+      data.autoPlayNextVideo = false;
+    default:
+      // Toggle borders off if user previously selected `related` type. `sponsored` and `editorial` types share defaults
+      data.playerBorderTop = data.previousTypeRelated ? false : data.playerBorderTop;
+      data.playerBorderBottom = data.previousTypeRelated ? false : data.playerBorderBottom;
+      data.previousTypeRelated = false;
+  }
+}
+
+function clearVideoId(data) {
+  data.videoId = (data.videoId || '').split('&')[0];
+
+  return data;
+}
+
+function setVideoDetails(data, videoDetails) {
+  var maxResThumb;
+
+  if (!videoDetails.title) {
+    data.videoValid = false;
+
+    return data;
+  }
+
+  maxResThumb = _get(videoDetails, 'thumbnails.maxres.url');
```

Comment:
We could declare this property here, instead of doing it in line 38
```suggestion
  const maxResThumb = _get(videoDetails, 'thumbnails.maxres.url'); // Remember to remove line 38
```